CppCoreGuidelines.md

```diff
@@ -13791,168 +13793,174 @@ Performance rule summary:
 
-???
+When you choose to rewrite a piece of code to increase its performance, do so
+because one of your important metrics is telling you to do so.  Check these
+metrics often, even preliminary versions, to make sure you're on track.  You can
+do this quickly and act before something very bad has happened.  "It looks good
+so far" is a great place to be. You do not want to get to your finish line only
+to discover that something is horribly wrong and you have to diagnose it as a
+whole.
 
-### <a name="Rper-critical"></a>Per.3: Don't optimize something that's not performance critical
+You do not have to boil out every last percentage immediately, you can often go
+wrong trying to do this too soon.  But you also do not want to be off by orders
+of magnitude, using an entirely inappropriate technique, a dependency you can't
+possibly afford, or completely disregarding a key metric either.  Regularly 
+verifying that you are on track is never premature.
 
-##### Reason
-
-Optimizing a non-performance-critical part of a program has no effect on system performance.
+Don Knuth popularized this quote by Tony Hoare, in full "We should forget about
+small efficiencies, say about 97% of the time: premature optimization is the
+root of all evil."
 
-##### Note
+[Small efficiencies can wait 97% of the time](https://ubiquity.acm.org/article.cfm?id=1513451#:~:text=Trying%20to%20do%20the%20optimization,best%20practice%20among%20software%20engineers.).
+Reasonable choices that give you a good chance of success cannot wait.  If you
+believe Sir Hoare -- which is wise -- 3% of the time small choices will burn you. That's
+about 1 or 2 per screenful of code.
 
-If your program spends most of its time waiting for the web or for a human, optimization of in-memory computation is probably useless.
 
-Put another way: If your program spends 4% of its processing time doing
-computation A and 40% of its time doing computation B, a 50% improvement on A is
-only as impactful as a 5% improvement on B. (If you don't even know how much
-time is spent on A or B, see <a href="#Rper-reason">Per.1</a> and <a
-href="#Rper-Knuth">Per.2</a>.)
 
-### <a name="Rper-simple"></a>Per.4: Don't assume that complicated code is necessarily faster than simple code
+### <a name="Rper-critical"></a>Per.3: Consider the causes of performance problems, not the effects.
 
 ##### Reason
 
-Simple code can be very fast. Optimizers sometimes do marvels with simple code
+Time is secondary metric.  Large time values are an effect of either bad
+orchestration, large resource consumption (read too much, use an O(N^2)
+algorithm instead of an O(N) alorigthm), or inefficient use of a resource (i.e.
+asking it to do something it's bad at, such as seeking all over, or dealing with
+non-local memory access).  Typically secondary metrics like time are not
+probative -- "the code is slow, boo hoo" -- primary metrics like "how many
```

Comment:
Probative is not a common word, I would find an alternative way to say this that will be more easily understood.

Reply:
reworded